app/notification.go

```diff
@@ -628,3 +606,49 @@ func (a *App) GetMentionKeywordsInChannel(profiles map[string]*model.User, lookF
 
 	return keywords
 }
+
+// Represents either an email or push notification and contains the fields required to send it to any user.
+type postNotification struct {
+	channel    *model.Channel
+	post       *model.Post
+	profileMap map[string]*model.User
+	sender     *model.User
+}
+
+// Returns the name of the channel for this notification. For direct messages, this is the sender's name
+// preceeded by an at sign. For group messages, this is a comma-separated list of the members of the
+// channel, with an option to exclude the recipient of the message from that list.
+func (n *postNotification) GetChannelName(userNameFormat string, excludeId string) string {
+	if n.channel.Type == model.CHANNEL_DIRECT {
+		return fmt.Sprintf("@%s", n.sender.GetDisplayName(userNameFormat))
+	} else if n.channel.Type == model.CHANNEL_GROUP {
```

Comment:
I can change it to a switch

I know else isn't needed, but I personally prefer that related if cases are done with else-if since I think it makes it easier to follow what's happening. Like I group these because they're both checking the channel type, but I separate unrelated ones like
```go
if channel.Type == model.CHANNEL_DIRECT {
    return channel.Name
}

if userNameFormat == "username" {
    return user.Username
}
```